### PR TITLE
Pivot table: column options

### PR DESCRIPTION
**Pull request by ariya**
Obviously this is still UI only for now, also with the messy styling (among others, I also lost the hover effect), but let me know if this is generally in the right direction.

image

### CODE REVIEW

File: frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPartition.jsx [hunk 1]
```diff
@@ -213,25 +265,56 @@ class EmptyPartition extends React.Component {
...
     return connectDropTarget(
       connectDragSource(
-        <div>
-          <ColumnDragger
+        <div
+          className={cx("mb1 bordered rounded")}
+          style={{
+            padding: "12px 14px",
+            "box-shadow": `0 2px 3px ${lighten(colors["text-dark"], 1.5)}`,
+            "&:hover": {
```
Comment (ariya) [inline frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPartition.jsx:292]: Looks like this doesn't work. If anyone has any idea, I'm all ears!

**Comment by mazameli** [comment]
I pushed some styling changes as well as some tweaks to the clickable vs. draggable areas, and did a bit of markup/css cleanup.

pivot-table-option-updates

**Review by paulrosenzweig — COMMENTED** [review]
Looks great so far! Thanks for jumping in!

Should we create issues for the settings that don't exist yet (hiding subtotals and sort order)? Maybe this could be merged with that UI hidden? Alternatively, we could make a mini feature branch to hold all this work together.

File: frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPartition.jsx [hunk 2]
```diff
@@ -1,22 +1,90 @@
...
+  }
+  toggleTotals = () => {
+    const { showTotals } = this.state;
+    this.setState({ showTotals: !showTotals });
+  };
+  render() {
+    const { showTotals } = this.state;
+    return (
+      <Flex pt={2} justifyContent="space-between" alignItems="center">
+        <Text>{t`Show totals`}</Text>
```
[the remark this answers is not in the document]
Comment (mazameli) [reply frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPartition.jsx:28]: Yeah, but I'm going to need to do one more trip around the POEditor block anyway 'cause of the cloud migration CTAs we added.

**Comment by mazameli** [comment]
> Should we create issues for the settings that don't exist yet (hiding subtotals and sort order)? Maybe this could be merged with that UI hidden?

My vote is to merge this in once the Formatting link is working, with the subtotals and sort order settings hidden. That de-risks things but also gets the new strings in faster so we can translate them. And agree we should have separate issues for the subtotal toggle and sort order; I'll create those.

Between those two remaining items, **sort order** is way more important than hiding subtotals.

File: frontend/src/metabase/visualizations/visualizations/PivotTable.jsx [hunk 3]
```diff
@@ -147,9 +147,11 @@ export default class PivotTable extends Component {
             columns = [first, second];
             rows = rest;
           }
-          setting = _.mapObject({ rows, columns, values }, cols =>
+          const fields = _.mapObject({ rows, columns, values }, cols =>
             cols.map(col => col.field_ref),
           );
+          const config = { answer: 42 };
```
Comment (ariya) [inline frontend/src/metabase/visualizations/visualizations/PivotTable.jsx:153]: This is just a sample (stub) config for now.

**Comment by ariya** [comment]
Note: I rebased to the latest master branch and looks like some tests are failing (not sure why, need to investigate).

**Comment by ariya** [comment]
> Note: I rebased to the latest master branch and looks like some tests are failing (not sure why, need to investigate).

I found the issue: it's because my change to add more config object into `COLUMN_SPLIT_SETTING`.

I took it away for now. Let's perhaps a different key for storing those (just like there is also `COLLAPSED_ROWS_SETTING` in addition to `COLUMN_SPLIT_SETTING`), so maybe this third one could be `FIELD_CONFIG_SETTING` or something along that line.

**Review by paulrosenzweig — APPROVED** [review]

**Comment by paulrosenzweig** [comment]
I moved things around a bit and connected the formatting options to the pivot UI. A remaining quirk is that columns in the body or top header don't display a column title, but still have the formatting setting. I think that's ok though, since as you move columns around you might not want the column title to clear itself.

I updated the arguments to `multiLevelPivot` which broke all the unit tests. Instead of the correct solution of adding helpers to create the (now required) settings, I split out a `multiLevelPivotForIndexes` that understand the arguments the test passes (raw indexes rather than field refs). I'll fix this up when I update those tests for sort order.

**Pull request merged**